src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1219,215 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
+                                "END '\n" +  
+                                "EXEC sp_MSforeachdb @script\n";
+                            // This query fetches all the notebook Jobs across all the databases accessible by the user.
+                            
+                        var agentNotebooks = new List<AgentNotebookInfo>();
+                        DataSet jobIdsDataSet = executeQuery(connInfo, getJobIdsFromDatabaseQueryString);
+                        var serverConnection = ConnectionService.OpenServerConnection(connInfo);
+                        var fetcher = new JobFetcher(serverConnection);
+                        var filter = new JobActivityFilter();
+                        // Fetching all the jobs
+                        var jobs = fetcher.FetchJobs(filter);
+                        // Creating Dictionary of jobs for effcient access later on.
+                        Dictionary<Guid, JobProperties> allJobsHashTable = new Dictionary<Guid, JobProperties>();
+                        if(jobs != null)
+                        {
+                            foreach (var job in jobs.Values)
+                            {
+                                allJobsHashTable.Add(job.JobID, job);
+                            }
+                        }                                
+                        foreach (DataTable templateTable in jobIdsDataSet.Tables)
+                        {
+                            foreach (DataRow templateRow in templateTable.Rows){
+                                AgentNotebookInfo notebookJob = AgentUtilities.convertToAgentNotebookInfo(allJobsHashTable[(Guid)templateRow["job_id"]]);
+                                notebookJob.TemplateId = templateRow["template_id"] as string;
+                                notebookJob.TargetDatabase = templateRow["db_name"] as string;
+                                agentNotebooks.Add(notebookJob);
+                            }
+                        }
+                    
+                        result.Success = true;
+                        result.Notebooks = agentNotebooks.ToArray();
+                        await requestContext.SendResult(result);
+                    }
+                }
+                catch (Exception e)
+                {
+                    await requestContext.SendError(e);
+                }
+            });
+        }
+
+        internal async Task HandleAgentNotebookHistoryRequest(AgentNotebookHistoryParams parameters, RequestContext<AgentNotebookHistoryResult> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                try
+                {
+                    var result = new AgentNotebookHistoryResult();
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                        parameters.OwnerUri,
+                        out connInfo);
+                    if (connInfo != null)
+                    {
```

Comment:
If connInfo is null you never send a response and thus will time out on the ADS side. You should send an error in that case.